Improved algorithm with OpenSSL 3.0.12 and 3.1.4

Sreekala-Gopakumar opened the pull request:
https://github.com/eclipse-openj9/openj9-website-publish/pull/6

Added the information related to improved algorithm with the use of OpenSSL version 3 and later

Signed-off-by: Sreekala Gopakumar <sreekala.gopakumar@ibm.com>

Sreekala-Gopakumar commented:
Jenkins doc stage

pshipton commented:
Same comment as https://github.com/eclipse-openj9/openj9-website-publish/pull/6#issuecomment-1783221915, pls add the "or later".

Sreekala-Gopakumar commented:
Jenkins doc stage

Sreekala-Gopakumar commented:
Jenkins doc stage

Sreekala-Gopakumar commented:
@pshipton - I have incorporated the suggested change. Please check and confirm. Thanks!

pshipton commented:
Sorry, I should have looked closer. This update should go into the What's New rather than the OpenSSL section. It's not the switch to OpenSSL v3 which provided perf benefits, this switch originally regressed perf. However there were a number of changes made to improve the performance with OpenSSL v3. Previous user's of OpenSSL v3 on Linux will see these improvements when 0.41 is used. For the best performance the fix added to OpenSSL 3.0.12 (3.1.4) should also be used.

Sreekala-Gopakumar commented:
> Sorry, I should have looked closer. This update should go into the What's New rather than the OpenSSL section. It's not the switch to OpenSSL v3 which provided perf benefits, this switch originally regressed perf. However there were a number of changes made in 0.41 to improve the performance with OpenSSL v3. Previous user's of OpenSSL v3 on Linux will see these improvements when 0.41 is used. For the best performance the fix added to OpenSSL 3.0.12 (3.1.4) should also be used.

So I should delete the newly added content from the openssl topic? 
> Performance of algorithms is enhanced with the use of OpenSSL version 3 and later. Improved algorithms include SHA256, AES, HmacSHA256, and ChaCha20. To obtain these improved algorithms on Linux and AIX, use OpenSSL 3.0.12 or a later 3.0.x version, or 3.1.4 or a later 3.1.x version because OpenSSL is not bundled with the Semeru JDK on those platforms.

This should be now added to the What's new in version 0.41.0?

pshipton commented:
> So I should delete the newly added content from the openssl topic?

If you mean the content added by this PR, then yes.

> This should be now added to the What's new in version 0.41.0? 

Correct.

> From your comments, it seems that the content itself needs a change. Should it be as follows?

I don't think we should state "You must use these versions", in particular the word "must". They are not required, they just provide the best v3 performance.

Sreekala-Gopakumar commented:
Jenkins doc stage

Sreekala-Gopakumar commented:
> > So I should delete the newly added content from the openssl topic?
> 
> If you mean the content added by this PR, then yes.
> 
> > This should be now added to the What's new in version 0.41.0?
> 
> Correct.
> 
> > From your comments, it seems that the content itself needs a change. Should it be as follows?
> 
> I don't think we should state "You must use these versions", in particular the word "must". They are not required, they just provide the best v3 performance.

I have deleted the last line "You must use these versions on Linux and AIX because OpenSSL is not bundled with the Semeru JDK on those platforms." because I think it is not necessary here. I hope that is ok. Thanks!
